lib/redisHandler.js

```diff
@@ -0,0 +1,231 @@
+const Resp = require('respjs');
+const EventEmitter = require('events');
+
+/**
+ * Class to handle a redis connection and provide events to react on for
+ * all incoming Redis commands
+ */
+class RedisHandler extends EventEmitter {
+    /**
+     * Initialize and register all data handlers to send out events on new commands
+     * @param socket Network Socket/Connection
+     * @param options options objects, currently mainly for logger
+     */
+    constructor(socket, options) {
+        super();
+
+        options = options || {};
+        this.log = options.log || console;
+        this.socket = socket;
+
+        this.sockedId = socket.remoteAddress + ':' + socket.remotePort;
+        this.initialized = false;
+
+        this.writeQueue = [];
+
+        this.resp = new Resp();
+
+        this.resp.on('error', (err) => {
+            this.log.error(this.sockedId + ' (Init=' + this.initialized + ') Redis error:' + err);
+            if (this.initialized) {
+                this.sendError(null,new Error('PARSER ERROR ' + err)); // TODO
+            }
+            else {
+                this.close();
+            }
+        });
+
+        this.resp.on('data', (data) => {
+            const command = data.splice(0, 1)[0];
+            if (command === 'info') this.initialized = true;
+            const t = process.hrtime();
+            const responseId = (t[0] * 1e3) + (t[1] / 1e6);
+            this.log.silly(this.sockedId + ' Parser result: id=' + responseId + ', command=' + command + ', data=' + JSON.stringify(data)); // TODO remove data logging (performance)
+            this.writeQueue.push({id: responseId, data: false});
+            if (this.listenerCount(command) !== 0) {
+                this.emit(command, data, responseId);
+            }
+            else {
+                this.sendError(responseId, new Error(command + 'NOT SUPPORTED'));
+            }
+        });
+        this.resp.on('finish', () => {
+            this.log.debug(this.sockedId + ' Redis finish');
+        });
+
+        socket.on('data', (data) => {
+            if (!data) return;
```

Comment:
Is it even possible that this is called with null or undefined?

Reply:
Ok, do not seem so ... The argument data will be a Buffer or String.